f8a_jobs/graph_ingestion.py

```diff
@@ -34,25 +39,32 @@ def ingest_epv_into_graph(epv_details):
           "force": false,              (optional)
           "force_graph_sync": true,    (optional)
           "recursive_limit": 0         (optional)
+          "source": "<Consumer_of_API>"(optional)
         }
     """
-    try:
-        logger.info('graph_ingestion_:_ingest_epv_into_graph() is called.')
-        input_data = epv_details.get('body', {})
-
-        # Check if EPV ingestion is enabled.
-        if _INVOKE_API_WORKERS:
-            ecosystem = input_data.get('ecosystem')
-            force = input_data.get('force', True)
-            force_graph_sync = input_data.get('force_graph_sync', False)
-            recursive_limit = input_data.get('recursive_limit', 0)
-
-            # Check if requested ecosystem for ingestion is supported,
-            # if not then set an error message.
-            if ecosystem in _SUPPORTED_ECOSYSTEMS:
-                package_list = input_data.get('packages')
-                gh = GithubUtils()
+    logger.info('graph_ingestion_:_ingest_epv_into_graph() is called.')
+    input_data = epv_details.get('body', {})
+    source = input_data.get('source', '')
+
+    if source == 'api' and _DISABLE_UNKNOWN_PACKAGE_FLOW:
+        logger.debug('Unknown package ingestion is disabled.')
+        input_data['message'] = 'Unknown package ingestion is disabled.'
+        return input_data, 201
+
+    # Check if EPV ingestion is enabled.
+    if _INVOKE_API_WORKERS:
```

Comment:
return early to keep the block simple.
```suggestion
    if not _INVOKE_API_WORKERS:
       logger.debug('API worker is disabled.')
        input_data['error_message'] = 'Unsupported ecosystem.'
        return input_data, 201
```